src/services/preProcess.ts

```diff
@@ -346,8 +346,8 @@ namespace ts {
                 }
 
                 if (scanner.getToken() === SyntaxKind.TemplateHead) {
-                    const stack = [scanner.getToken()];
-                    let token = scanner.scan();
+                    let token = scanner.getToken();
+                    const stack = [token];

```

Comment:
If the starting token is `TemplateHead`, then we get down to the switch below and then it will push it onto the stack a second time. It seems like the old code was correct, so I'm not sure what I'm missing here that would make this change fix the bug...

Reply:
oke

Reply:
It seems we need to skip moving forward to the next token when the `stack` is empty.